diffyscan/utils/compiler.py

```diff
@@ -0,0 +1,109 @@
+import platform
+import hashlib
+import subprocess
+import json
+import os
+import stat
+import sys
+
+from .common import fetch
+from .helpers import create_dirs
+from .logger import logger
+
+
+def get_solc_native_platform_from_os():
+    platform_name = sys.platform
+    if platform_name == "linux":
+        return "linux-amd64"
+    elif platform_name == "darwin":
+        return "macosx-amd64" if platform.machine() == "x86_64" else "macosx-arm64"
+    else:
+        raise ValueError(f"Unsupported platform {platform_name}")
+
+
+def get_compiler_info(platform, required_compiler_version):
```

Comment:
platform shadows the import platform 
```suggestion
def get_compiler_info(required_platform, required_compiler_version):
```

Reply:
done